### ISSUE TITLE
No need to nVal in Shape Data.

### ISSUE THREAD

**Issue by AniruddhaHumane** [gbouritsas/Neural3DMM#8]
`nVal` in `ShapeData` serves no purpose. It just reduces training data by 100 samples.

**Comment by gbouritsas** [comment]
Hi, if I recall correctly you are referring to the validation set of the COMA dataset, right? This is a standard benchmark and the validation set can be used for hyperparameter tuning, early stopping etc.

**Comment by AniruddhaHumane** [comment]
However, isn't the validation set formed by `data_generation.py`? The `autoencoder_dataset` then uses the paths and point files created by this script as training, validation and test set. The validation set inside `shape_data` is not used for validation.

**Comment by gbouritsas** [comment]
I checked the code now - I believe we use nVal only when we construct a ShapeData object. 

When we initialise it, we need to calculate the normalisation parameters (mean and std). These should be computed on the training set only. Hence, we need to split the training data at this point, to avoid taking into account validation data in the computation of the aforementioned parameters. Does this help?

**Comment by AniruddhaHumane** [comment]
Oh right! It makes sense now since `shapedata` uses raw inputs and not the output generated by `data_generation.py`. Thanks @gbouritsas 😄
